doc/dataModel.md

```diff
@@ -0,0 +1,242 @@
+                },
+                "metadata": [
+                    {
+                        "type": "required",
+                        "scope": [
+                            "http://registry.gpii.net/common/screenReaderTTSEnabled"
+                        ],
+                        "value": 1024
+                    },
+                    {
+                        "type": "priority",
+                        "scope": [
+                            "http://registry.gpii.net/applications/com.freedomScientific.jaws"
+                        ],
+                        "value": 1024
+                    }
+                ]
+            },
+            "internalID-1": {
+                "name": "subway",   // user defined name
+                "preferences": {
+                    "http://registry.gpii.net/common/highContrastEnabled": true,
+                    "http://registry.gpii.net/common/highContrastTheme": "white-black",
+                    "http://registry.gpii.net/common/cursorSize": 0.5
+                },
+                "metadata": [
+                    {
+                        "type": "required",
+                        "scope": [
+                            "http://registry.gpii.net/common/screenReaderTTSEnabled"
+                        ],
+                        "value": 1024
+                    },
+                    {
+                        "type": "priority",
+                        "scope": [
+                            "http://registry.gpii.net/applications/com.freedomScientific.jaws"
+                        ],
+                        "value": 1024
+                    }
+                ],
+                "conditions": [
+                    {
+                        "type": "http://registry.gpii.net/conditions/inRange",
+                        "min": 400,
+                        "inputPath": "http://registry\\.gpii\\.net/common/environment/illuminance"
+                    }
+                ]
+            }
+        }
+    }
+    "ISO24751": {
+        "contexts": {
+            "gpii-default": {
+                "name": "Default preferences",
+                "preferences": {
+                    "display": {
+                        "screenEnhancement": {
+                            "fontSize": 24
+                        }
+                    },
+                    "control": {
+                        "onscreenKeyboard": true
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+### clientCredentials
+
+This table contains all OAuth2 client credentials assigned to all OAuth clients, including: 1. GPII app installation clients;
+2. Privileged prefs creator clients; 3. Web prefs consumer clients.
+
+* At any time, each client should only have one active un-revoked client credential in this document.
+* TODO: there's an ongoing discussion on whether to stop support privileged prefs creator clients and web prefs consumer clients.
```

Comment:
I suggest adding a general TODO section since there are a number of other aspects of the data model that will likely change.  Here are some more:
- TODO: remove ontologies
- TODO: remove/replace all occurrences of `gpii` and `GPII`.
  - change `GPII Keys` data model to `Keys` (?)
  - change the type of a `Key` from `gpiiKey` to `key`(?)
  - change `GPII App Installation Client` to `App Installation Client`, 
  - change its `type` from `gpiiAppInstallationClient` to `appInstallationClient`
  - change `GPII App  Installation Authorization` to `App Installation Authorization`
  - change its `type` `gpiiAppInstallationAuthorization` to `appInstallationAuthorization`
  - change the `type` `gpiiExpressUserId` to `expressUserId`, or just `userId`.  How important is it that it is an express user?
 - TODO: change `contexts`, the top level within an ontology, to `prefsSets`, representing a container of named sets of preferences.

Reply:
In terms of renaming `GPII Keys` to `Keys` and `gpiiKey` to `key`, in my personal opinion, the use of`key` or `keys` are ambiguous especially in the context of database world where terms like primary keys or foreign keys are heavily used. I would suggest when we are clear about how these keys will be used in the workflow, rename these to more specific names that can easily differentiate them from the common use of "keys'.

Reply:
In the GPII project, keys are used to identify computers or on USBs for users' convenient access to their preferences without extra interactions like the login. I'm not sure if this use case will continue to be supported in this prefs server work. The use case we've talked so far is users will login to get access to the prefs server. If this is the only use case the prefs server will support, "keys" document can be removed.

Reply:
All right.  In that case, I suggest changing the TODO to:
- Change the name of GPII Keys document or, if it is no longer a use case, remove it entirely